earthfile2llb/converter.go

```diff
@@ -550,6 +550,9 @@ func (c *Converter) RunExitCode(ctx context.Context, opts ConvertRunOpts) (int,
 			return 0, errors.Wrap(err, "read exit code file")
 		}
 	} else {
+		if c.opt.NoCache {
+			state = state.SetMarshalDefaults(llb.IgnoreCache)
+		}
```

Comment:
I would maybe add a `noCache` parameter to `StateToRef` and pass in `c.opt.NoCache` for that in all these places. That should catch all other similar places that have this issue: `FOR`, dynamic args, `WITH DOCKER` etc.